framework/caching/Cache.php

```diff
@@ -358,9 +389,11 @@ public function add($key, $value, $duration = 0, $dependency = null)
         } elseif ($this->serializer !== false) {
             $value = call_user_func($this->serializer[0], [$value, $dependency]);
         }
-        $key = $this->buildKey($key);
 
-        return $this->addValue($key, $value, $duration);
+        $addResult = $this->addValue($builtKey, $value, $duration);
+        $this->endProfile($builtKey, __METHOD__);
```

Comment:
same as above, profiling should only be done for the cache interaction, not the serializaion to be consistent with the other methods.